dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/artifactbrowser/ItemViewer.java

```diff
@@ -453,4 +465,104 @@ public void recycle() {
     	this.validity = null;
     	super.recycle();
     }
+
+    private static class GoogleDataset{
+        private static final Set<String> mandatoryFields = new HashSet<>();
+        private static final HashMap<String, String> google2dspace = new HashMap<>();
+        static {
+            mandatoryFields.add("name");
+            mandatoryFields.add("description");
+
+            google2dspace.put("name", "dc.title");
+            google2dspace.put("description", "dc.description");
+
+            String pathToProps = new DSpace().getConfigurationService().getPropertyAsType("google-metadata.config", "");
+            if(!pathToProps.trim().isEmpty()) {
+                Properties properties = new Properties();
+                try {
+                    BufferedReader reader = Files.newBufferedReader(Paths.get(pathToProps), StandardCharsets.UTF_8);
+                    properties.load(reader);
+                } catch (IOException e) {
+                    log.error("Exception while reading google-metadata.config", e);
+                }
+                for (String key : properties.stringPropertyNames()) {
+                    if (key.startsWith("google.dataset_")) {
+                        String googleKey = key.replace("google.dataset_", "").trim();
+                        google2dspace.put(googleKey, properties.getProperty(key).trim());
+                    }
+                }
+            }
+        }
+
+
+        private JsonObject jsonObject;
+        GoogleDataset(Item item){
+            jsonObject = new JsonObject();
+            jsonObject.addProperty("@context", "https://schema.org");
+            jsonObject.addProperty("@type", "Dataset");
+            for(Entry<String, String> entry : google2dspace.entrySet()){
+                String googleKey = entry.getKey();
+                String dspaceField = entry.getValue();
+                Metadatum[] metadata = item.getMetadataByMetadataString(dspaceField);
+                if(metadata.length == 1){
+                    jsonObject.addProperty(googleKey, metadata[0].value);
+                }else if(metadata.length > 1){
+                    JsonArray values = new JsonArray();
+                    for(Metadatum md : metadata){
+                        values.add(new JsonPrimitive(md.value));
+                    }
+                    jsonObject.add(googleKey, values);
+                }else if (mandatoryFields.contains(googleKey)){
+                    jsonObject.addProperty(googleKey, "(:unav)");
+                }else{
+                    continue;
+                }
+            }
+            convertToValidValues(jsonObject);
+        }
+
+        private void convertToValidValues(JsonObject jsonObject) {
+            String description = jsonObject.get("description").getAsString();
+            if(description.length() < 50){
```

Comment:
minor - magic values (50,5000)

Reply:
see the comment below (or https://developers.google.com/search/docs/data-types/dataset#dataset).
I can extract them into variables/constants with more reasonable meaning (like `minDescriptionLength`, `maxDescriptionLength`) if that's what you are suggesting.

Reply:
@kosarko exactly